File: instautils/src/main/java/com/sanjaydevtech/instautils/InstaResponse.java
package com.sanjaydevtech.instautils;

/**
 * InstaResponse to handle download url searching
 */
public interface InstaResponse {
    /**
     * On successful url searching
     *
     * @param post InstaPost object
     */
    void onResponse(InstaPost post);

    /**
     * When an error occurs
     *
     * @param e Exception
     */
    void onError(Exception e);
}

File: instautils/src/main/java/com/sanjaydevtech/instautils/InstaResponse.kt
package com.sanjaydevtech.instautils

/**
 * InstaResponse to handle download url searching
 */
fun interface InstaResponse {
    /**
     * On completion of url searching
     *
     * @param instaTask InstaTask object which holds the post or exception
     */
    fun onResponse(instaTask: InstaTask)
}